harness/determined/common/schemas/expconf/_v0.py

```diff
@@ -224,44 +224,56 @@ def __init__(
 class EnvironmentImageV0(schemas.SchemaBase):
     _id = "http://determined.ai/schemas/expconf/v0/environment-image.json"
     cpu: Optional[str] = None
-    gpu: Optional[str] = None
+    cuda: Optional[str] = None
+    rocm: Optional[str] = None
 
     @schemas.auto_init
     def __init__(
         self,
         cpu: Optional[str] = None,
-        gpu: Optional[str] = None,
+        cuda: Optional[str] = None,
+        rocm: Optional[str] = None,
     ) -> None:
         pass
 
     @classmethod
     def from_dict(cls, d: Union[dict, str], prevalidated: bool = False) -> "EnvironmentImageV0":
         # Accept either a string or a map of strings to strings.
         if isinstance(d, str):
-            d = {"cpu": d, "gpu": d}
-        return super().from_dict(d, prevalidated)
+            d = {"cpu": d, "cuda": d, "rocm": d}
+        if "cuda" not in d and "gpu" in d:
+            d["cuda"] = d["gpu"]
+            del d["gpu"]
```

Comment:
Non-blocking:

Would this be a more desirable migration strategy:
```python
if "gpu" in d:
    d["cuda"] = d["gpu"]
    d["rocm"] = d["gpu"]
    del d["gpu"]
```

Pros I can think of:
- forking an existing experiment where the user original set `image: alpine` (which we converted to `cpu: alpine`/`gpu: alpine`) will still render with `rocm: alpine`
 
Cons I can think of:
- `gpu` will have a unique behavior that some users might grow to depend on, making it potentially hard to remove later.
- when forking, using an old `gpu` default image as the `rocm` image is probably unhelpful.

**Update**: I think the last con has convinced me that I like what you have best, but I'm still posting this for your consideration.

Reply:
>when forking, using an old gpu default image as the rocm image is probably unhelpful.
right. I also don't see how that can ever work.